### PR TITLE
Make the linter print better error messages

### LINKED ISSUE #2143: Investigate use of better-ajv-errors
The goal of https://www.npmjs.com/package/better-ajv-errors is to provide better error messages when schema validation fails. We should investigate if integrating this package improves our typical error messages.

### LINKED ISSUE #2330: Non-informative messages about JSON errors
I left trailing comma in https://github.com/mdn/browser-compat-data/pull/2329/commits/9acd41646c210dc0598fd2ed8d6a5d9d99851e2d but Travis reported something weird
https://travis-ci.org/mdn/browser-compat-data/builds/393218140#L827

Can linter be replaced with something more user friendly? If I didn't knew about trailing commas problem in JSON, I could probably spend an hour trying to hunt it down.

### PR DESCRIPTION
**Pull request by ExE-Boss**
Depends on #2288

Fixes #2330
Fixes #2143 

---

[Example result with a failed build](https://travis-ci.org/mdn/browser-compat-data/builds/394198004#L7821)

### CODE REVIEW

**Comment by ExE-Boss** [comment]
@Elchi3 I’ve now rebased this and #2155.

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
   "homepage": "https://github.com/mdn/browser-compat-data#readme",
   "devDependencies": {
     "ajv": "^5.0.1",
+    "better-ajv-errors": "^0.5.1",
```
[the remark this answers is not in the document]
Comment (ExE-Boss) [reply package.json]: 0.6.0 wasn’t out yet when I made this.

Anyway, I’ll go ahead and update this.

**Comment by ExE-Boss** [comment]
review?(@Elchi3): I have now resolved all the merge conflicts with the `master` branch and updated `ajv` and `better‑ajv‑errors` to the latest versions.

Also, `package‑lock.json` was very annoying, as I use [pnpm](https://pnpm.js.org) for my day‑to‑day node package development, but `npm ci` requires that `package.json` and `package‑lock.json` match.

File: test/lint.js [hunk 2]
```diff
@@ -71,7 +79,7 @@ function load(...files) {
         }
       }
 
-      continue;
+      return;
```
Comment (Elchi3) [inline test/lint.js]: Why was this changed to `return`? 
I think we want to continue with more files, otherwise you only get the first file per folder.
See the most recent travis build: https://travis-ci.org/mdn/browser-compat-data/builds/472649710

File: test/test-schema.js [hunk 3]
```diff
@@ -1,23 +1,24 @@
 'use strict';
 const Ajv = require('ajv');
+const betterAjvErrors = require('better-ajv-errors');
 const path = require('path');
-const ajv = new Ajv({ allErrors: true });
+
+const ajv = new Ajv({ jsonPointers: true, allErrors: true });
```
Comment (Elchi3) [inline test/test-schema.js]: Why was this added? `jsonPointers: true,` ?

Comment (ExE-Boss) [reply test/test-schema.js]: Because `better‑ajv‑errors` requires it&nbsp;to&nbsp;function.

Comment (Elchi3) [reply test/test-schema.js]: Ah okay, cool! :+1:

File: test/lint.js (hunk 2 above)
Comment (ExE-Boss) [reply test/lint.js]: I accidentally screwed up a git merge.
```suggestion
      continue;
```
